### PR TITLE
Start BEEFY test support lib

### PR DESCRIPTION
**Pull request by adoerr**
Introduce a BEEFY test support library. 

The idea for this lib is to bundle all the test related functionality, so that it can be referenced as one dev-dependency. Also the light client test implementation should be moved. here.

### CODE REVIEW

File: beefy-test/src/keyring.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
```
Comment (tomusdrw) [inline beefy-test/src/keyring.rs:16]:
```suggestion

#![warn(missing_docs)]

```

File: beefy-test/src/keyring.rs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		format!("//{}", self)
+	}
+
+	/// Iterator over all test accounts
+	pub fn iter() -> impl (Iterator<Item = Keyring>) {
+		<Self as strum::IntoEnumIterator>::iter()
+	}
+}
+
+impl From<Keyring> for &'static str {
```
Comment (tomusdrw) [inline beefy-test/src/keyring.rs:65]: Why is that useful / needed?

Comment (adoerr) [reply beefy-test/src/keyring.rs:65]: It is used for `Keyring::pair()`

Comment (tomusdrw) [reply beefy-test/src/keyring.rs:67]: But we also have `to_seed` and it's using `strum::Display`? Why do we really need both and risk getting mismatch?

Comment (adoerr) [reply beefy-test/src/keyring.rs]: Changed the `Keyring::pair()` implementation.

**Review by tomusdrw — APPROVED** [review]

**Pull request merged**